aptos-move/dynamic-transaction-composer/src/lib.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright © Aptos Foundation
+// Parts of the project are originally copyright © Meta Platforms, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+pub use crate::{
+    builder::TransactionComposer,
+    decompiler::{generate_batched_call_payload, generate_batched_call_payload_wasm},
+};
+use move_core_types::{
+    identifier::Identifier,
+    language_storage::{ModuleId, TypeTag},
+};
+use serde::{Deserialize, Serialize};
+use tsify_next::Tsify;
+use wasm_bindgen::prelude::wasm_bindgen;
+
+mod builder;
+mod decompiler;
+mod helpers;
+
+#[cfg(test)]
+pub mod tests;
+
+// CompiledScript generated from script builder will have this key in its metadata to distinguish from other scripts.
+pub static APTOS_SCRIPT_BUILDER_KEY: &[u8] = "aptos::script_builder".as_bytes();
+
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct PreviousResult {
+    call_idx: u16,
+    return_idx: u16,
+    operation_type: ArgumentOperation,
+}
+
+#[derive(Clone, Debug, Hash, Eq, PartialEq, Serialize, Deserialize, Tsify)]
+#[tsify(into_wasm_abi, from_wasm_abi)]
+pub enum BatchArgument {
```

Comment:
Yeah as we discussed offline, consider unifying `BatchArgument` and `Argument`

Reply:
Done